schema/schema.go

```diff
@@ -224,24 +224,19 @@ func copyDescriptions(schema *Schema, root string, defs map[string]Definition) {
 
 		path := normalizePath(fqName)
 		if path != "" && def.Description != "" {
-			if def.Type == "group" {
-				ns := schema.Namespaces[path]
-				if ns == nil {
-					panic(fmt.Sprintf("no namespace for: %v", path))
-				}
-
+			if ns := schema.Namespaces[path]; ns != nil {
 				ns.Description = def.Description
-			} else {
-				val, ok := schema.Values[path]
-				if !ok {
-					continue
-				}
-				if val == nil {
-					panic(fmt.Sprintf("no value for: %v", path))
-				}
+			} else if def.Type == "group" {
+				panic(fmt.Sprintf("no namespace for: %v", path))
+			}
+
+			if val, ok := schema.Values[path]; ok && val != nil {
 
 				val.Description = def.Description
+			} else if def.Type != "group" {
+				panic(fmt.Sprintf("no value for: %v", path))
```

Comment:
This changes semantics a little. Before we just continued processing if ok is false, no we panic if ok is false. Is there a reason for this change?

schema.Values should only be queried if it is not group. Given the change of conditions we always look into schema.Values if we found the path in schema.Namespaces. Is there a reason for this change? Normally `Namespaces` should have the entry if `Type == group`.

Reply:
The reason for this change is that for some entries that are `Namespaces` the type is `object`. The change ensures that we also have a description available for these types. Using the current generator logic e.g. in combination with the `ecs_nested.yml` leads to not having a description on some objects, which is an issue when the type should be exported and documentation auto generated. 

I agree that there is a slight change in semantics, but I couldn't think of a valid case where we just want to ignore if `ok` was `false`, therefore I changed to also panic in this case. IMO it is better to panic with the generator than to silently swallow issues. It allows to add more specific handling when the generator is in heavier use and evolves.